packages/compiler/core/checker.ts

```diff
@@ -1009,35 +1034,130 @@ export function createChecker(program: Program): Checker {
   }
 
   function checkOperation(
-    node: OperationStatementNode,
+    node: OperationStatementNode | OperationInstanceNode,
     parentInterface?: InterfaceType
-  ): OperationType {
+  ): OperationType | ErrorType {
+    const links = getSymbolLinks(node.symbol);
+    const instantiatingThisTemplate = instantiatingTemplate === node;
+    if (links.declaredType && !instantiatingThisTemplate) {
+      // we're not instantiating this operation and we've already checked it
+      return links.declaredType as OperationType;
+    }
+
     const namespace = getParentNamespaceType(node);
     const name = node.id.sv;
     const decorators = checkDecorators(node);
-    const type: OperationType = createType({
+
+    // Is this a definition or instance?
+    let parameters: ModelType, returnType: Type;
+    if (node.kind === SyntaxKind.OperationInstance) {
+      // Attempt to resolve the operation
+      const baseOperation = checkOperationIs(node, node.baseOperation);
+      if (!baseOperation) {
+        // TODO: Are the proper diagnostics written already?
+        return errorType;
+      }
+
+      // Reference the same return type and create the parameters type
+      returnType = baseOperation.returnType;
+      parameters = createType({
+        kind: "Model",
+        name: "",
+        node: baseOperation.parameters.node, // TODO: This seems bad!
+        properties: new Map<string, ModelTypeProperty>(),
+        namespace: getParentNamespaceType(node),
+        decorators: [],
+        derivedModels: [],
+      });
+
+      // Copy parameters of the base operation
+      for (const prop of baseOperation.parameters.properties.values()) {
+        console.log("copying param:", prop.name, (prop.type as any).kind);
+        // Don't use the same property, clone it and finish it to execute the decorators again
+        parameters.properties.set(
+          prop.name,
+          finishType({
+            ...prop,
+          })
+        );
+      }
+    } else {
+      parameters = getTypeForNode(node.parameters) as ModelType;
+      returnType = getTypeForNode(node.returnType);
+    }
+
+    const operationType: OperationType = createType({
       kind: "Operation",
       name,
       namespace,
       node,
-      parameters: getTypeForNode(node.parameters) as ModelType,
-      returnType: getTypeForNode(node.returnType),
-      decorators,
+      parameters,
+      returnType,
+      decorators, // TODO: Concatenate base operation decorators recursively!
```

Comment:
I think you won't need to do any recursion here because the decorators on baseOperation will already be fully present including any clones from its own baseOperation?